volatility/framework/plugins/windows/hashdump.py

```diff
@@ -0,0 +1,343 @@
+        return des1.decrypt(obfkey[:8]) + des2.decrypt(obfkey[8:16])
+
+    @classmethod  
+    def get_user_hashes(cls, user, samhive, hbootkey):
+        ## Will sometimes find extra user with rid = NAMES, returns empty strings right now
+        try:
+            rid = int(str(user.get_name()), 16)
+        except ValueError:
+            return None
+        sam_data = None
+        for v in user.get_values():
+            if v.get_name() == 'V':
+                sam_data = samhive.read(v.Data+4, v.DataLength)
+        if not sam_data:
+            return None
+        
+        lm_offset = unpack("<L", sam_data[0x9c:0xa0])[0] + 0xCC
+        lm_len = unpack("<L", sam_data[0xa0:0xa4])[0]
+        nt_offset = unpack("<L", sam_data[0xa8:0xac])[0] + 0xCC
+        nt_len = unpack("<L", sam_data[0xac:0xb0])[0]
+
+
+        lm_revision = sam_data[lm_offset + 2:lm_offset + 3]
+        if lm_revision == b'\x01':
+            lm_exists = True if lm_len == 20 else False
+            enc_lm_hash = sam_data[lm_offset + 0x04:lm_offset + 0x14] if lm_exists else ""
+            lmhash = cls.decrypt_single_hash(rid, hbootkey, enc_lm_hash, cls.almpassword)
+        elif lm_revision == b'\x02':
+            lm_exists = True if lm_len == 56 else False
+            lm_salt = sam_data[hash_offset+4:hash_offset+20] if lm_exists else ""
+            enc_lm_hash = sam_data[hash_offset+20:hash_offset+52] if lm_exists else ""
+            lmhash = cls.decrypt_single_salted_hash(rid, hbootkey, enc_lm_hash, cls.almpassword, lm_salt)
+
+        # NT hash decryption
+        nt_len = unpack("<L", sam_data[0xac:0xb0])[0]
+
+        nt_revision = sam_data[nt_offset + 2:nt_offset + 3]
+        if nt_revision == b'\x01':
+            nt_exists = True if nt_len == 20 else False
+            enc_nt_hash = sam_data[nt_offset+4:nt_offset+20] if nt_exists else ""
+            nthash = cls.decrypt_single_hash(rid, hbootkey, enc_nt_hash, cls.antpassword)
+        elif nt_revision == b'\x02':
+            nt_exists = True if nt_len == 56 else False
+            nt_salt = sam_data[nt_offset+8:nt_offset+24] if nt_exists else ""
+            enc_nt_hash = sam_data[nt_offset+24:nt_offset+56] if nt_exists else ""
+            nthash = cls.decrypt_single_salted_hash(rid, hbootkey, enc_nt_hash, cls.antpassword, nt_salt)
+        return lmhash, nthash
+    
+    #Takes rid of a user and converts it to a key to be used by the DES cipher
+    @classmethod
+    def sid_to_key(cls, sid):
+
+        str1 = ""
+        str1 += chr(sid & 0xFF)
+        str1 += chr((sid >> 8) & 0xFF)
+        str1 += chr((sid >> 16) & 0xFF)
+        str1 += chr((sid >> 24) & 0xFF)
+        str1 += str1[0]
+        str1 += str1[1]
+        str1 += str1[2]
+        str2 = str1[3] + str1[0] + str1[1] + str1[2]
+        str2 += str2[0] + str2[1] + str2[2]
+        return cls.str_to_key(str1), cls.str_to_key(str2)
+
+    #Build final DES key from the strings generated in sid_to_key
```

Comment:
This would be better as a docstring, since that'll then get automatically added to the documentation we generate (https://volatility3.readthedocs.io/en/latest/volatility.plugins.windows.html#submodules)

So in this case it would be:
```
@classmethod
def str_to_key(cls, s):
    """Builds final DES key from the strings generated in sid_to_ley"""
```